src/TortoiseProc/Settings/SetLookAndFeelPage.cpp

```diff
@@ -32,23 +32,30 @@ extern MenuInfo menuInfo[];
 void InsertMenuItemToList(CListCtrl *list,CImageList *imagelist)
 {
 	int i=0;
+	HINSTANCE hRes = AfxGetResourceHandle();
 	while(menuInfo[i].command != ShellMenuLastEntry)
 	{
-		if ((menuInfo[i].command != ShellSeparator &&
-		   menuInfo[i].command != ShellSubMenu &&
-		   menuInfo[i].command != ShellSubMenuFile &&
-		   menuInfo[i].command != ShellSubMenuFolder &&
-		   menuInfo[i].command != ShellSubMenuLink &&
-		   menuInfo[i].command != ShellSubMenuMultiple) &&
-		   (i == 0 || menuInfo[i - 1].menuID != menuInfo[i].menuID))
+		const MenuInfo& mi = menuInfo[i];
+		if ((mi.command != ShellSeparator &&
+			mi.command != ShellSubMenu &&
+			mi.command != ShellSubMenuFile &&
+			mi.command != ShellSubMenuFolder &&
+			mi.command != ShellSubMenuLink &&
+			mi.command != ShellSubMenuMultiple) &&
+			(i == 0 || menuInfo[i - 1].menuID != mi.menuID))
 		{
-			HICON hIcon = reinterpret_cast<HICON>(::LoadImage(AfxGetResourceHandle(),
-					MAKEINTRESOURCE(menuInfo[i].iconID),IMAGE_ICON, 16, 16, LR_LOADTRANSPARENT ));
+			HICON hIcon = reinterpret_cast<HICON>(::LoadImage(

```

Comment:
You mean a helper function like this?

```
HICON LoadIconFromResource(WORD wInteger, bool isSmall, UINT fuLoad)
{
	int width = ::GetSystemMetrics(isSmall ? SM_CXSMICON : SM_CXICON);
	int height = ::GetSystemMetrics(isSmall ? SM_CYSMICON : SM_CYICON);
	return reinterpret_cast<HICON>(::LoadImage(AfxGetResourceHandle(),
					MAKEINTRESOURCE(wInteger),IMAGE_ICON, width, height, fuLoad));
}
```

By the way, I found interesting note about `LR_LOADTRANSPARENT` in this page.
https://msdn.microsoft.com/en-us/library/windows/desktop/ms648045(v=vs.85).aspx
> Do not use this option if you are loading a bitmap with a color depth greater than 8bpp.